core/templates/dev/head/pages/exploration_player/ProgressNavDirective.js

```diff
@@ -20,39 +20,69 @@ oppia.directive('progressNav', [
   'UrlInterpolationService', function(UrlInterpolationService) {
     return {
       restrict: 'E',
+      scope: {
+        onSubmit: '&',
+        onClickContinueButton: '&'
+      },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/exploration_player/' +
         'progress_nav_directive.html'),
       controller: [
         '$scope', '$rootScope', 'PlayerPositionService',
-        'PlayerTranscriptService',
+        'PlayerTranscriptService', 'ExplorationPlayerService',
+        'ExplorationPlayerStateService', 'CONTINUE_BUTTON_FOCUS_LABEL',
+        'INTERACTION_SPECS',
         function($scope, $rootScope, PlayerPositionService,
-          PlayerTranscriptService) {
-          $scope.currentCardIndex = PlayerPositionService.getActiveCardIndex();
-          $scope.transcriptLength = PlayerTranscriptService.getNumCards();
+          PlayerTranscriptService, ExplorationPlayerService,
+          ExplorationPlayerStateService, CONTINUE_BUTTON_FOCUS_LABEL,
+          INTERACTION_SPECS) {
+          $scope.CONTINUE_BUTTON_FOCUS_LABEL = CONTINUE_BUTTON_FOCUS_LABEL;
+
+          var transcriptLength, isInteractionInline, interaction;
```

Comment:
- Move `interaction` inside, since it's only needed within the scope of updateActiveCardInfo(). I.e. do "var interaction = ..." on line 51.
- Please rename isInteractionOutline to interactionIsInline so it doesn't start with a verb.
- Instantiate transcriptLength to 0, and interactionIsInline to true, on separate lines.